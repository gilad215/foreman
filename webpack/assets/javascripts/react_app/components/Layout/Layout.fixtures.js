const mockOnClick = jest.fn();

// PF-React Component Data

const PFitems = [{
  title: 'Monitor', initialActive: true, iconClass: 'fa fa-tachometer', subItems: subItemsA, href: '/',
},
{
  title: 'Hosts', initialActive: false, iconClass: 'fa fa-server', subItems: subItemsB, href: '/',
},
{
  title: 'Configure', initialActive: false, iconClass: 'fa fa-wrench', subItems: subItemsC, href: '/',
}];

const subItemsA = [{
  title: 'A', isDivider: false, onClick: mockOnClick,
},
{
  title: 'B', isDivider: true, onClick: mockOnClick,
},
{
  title: 'C', isDivider: false, onClick: mockOnClick,
},
];
const subItemsB = [{
  title: 'D', isDivider: false, onClick: mockOnClick,
},
{
  title: 'E', isDivider: true, onClick: mockOnClick,
},
{
  title: 'F', isDivider: false, onClick: mockOnClick,
},
];
const subItemsC = [{
  title: 'G', isDivider: false, onClick: mockOnClick,
},
{
  title: 'H', isDivider: true, onClick: mockOnClick,
},
{
  title: 'I', isDivider: false, onClick: mockOnClick,
},
];

// Server Hash Data
const monitorChildren = [{
  type: 'item', name: 'Dashboard', url: '/',
},
{ type: 'divider', name: 'Hello' },
{
  type: 'item', name: 'Facts', url: '/fact_values',
},
];

const hostsChildren = [{
  type: 'item', name: 'All Hosts', url: '/hosts/new',
},
{ type: 'divider', name: 'There' },
{
  type: 'item', name: 'Architectures', url: '/architectures',
},
];

const userChildren = [{
  type: 'item', name: 'Environments', url: '/environments',
},
{ type: 'divider', name: 'Dear' },
{
  type: 'item', name: 'Architectures', url: '/architectures',
},
];
const infrastructureChildren = [{
  type: 'item', name: 'Domains', url: '/domains',
},
{ type: 'divider', name: 'Friend' },
{
  type: 'item', name: 'Realms', url: '/realms',
},
];

const hashItemsA = [{
  type: 'sub_menu', name: 'Monitor', icon: 'fa fa-tachometer', children: monitorChildren, active: false,
},
{
  type: 'sub_menu', name: 'Hosts', icon: 'fa fa-server', children: hostsChildren, active: false,
}];

const hashItemsB = [{
  type: 'sub_menu', name: 'User', icon: 'fa fa-wrench', children: userChildren, active: false,
},
{
  type: 'sub_menu', name: 'Infrastructure', icon: 'pficon pficon-network', children: infrastructureChildren, active: false,
}];

export const serverItems = [...hashItemsA, ...hashItemsB];
// DATA PROPS

const logo = '/assets/header_logo-c9614c16f2ee399ae9cb7f36ec94b9a26bf8cf9eabaa7fe6099bf80d1f7940db.svg';
const user = {
  current_user: {
    user: {
      id: 4, login: 'admin', firstname: 'Admin', lastname: 'User',
    },
  },
  user_dropdown: [
    {
      type: 'sub_menu', name: 'User', icon: 'fa fa-user', children: subItemsA,
    },
  ],
};

const organizations = {
  current_org: {
    organization: { id: 1, name: 'org1' },
  },
  available_organizations: [{ id: 1, name: 'org1', href: '/organizations/1-org1/select' }],
};

const locations = {
  current_location: {
    location: { id: 1, name: 'yaml' },
  },
  available_locations: [{ id: 1, name: 'yaml', href: '/locations/1-yaml/select' }],
};

const serverUser = {
  current_user: {
    user: {
      firstname: 'G',
      lastname: 'L',
    },
  },
  user_dropdown: [{
    children: [{
      type: 'item', url: '/', name: 'My Account',
    }, { type: 'divider' },
    ],
  }],
};

export const layoutMock = {
  items: PFitems,
  data: {
    menu: [hashItemsA, hashItemsB],
    locations,
    organizations,
    root: '/',
    logo,
    notification_url: '/notification_recipients',
    taxonomies: { locations: true, organizations: true },
    user,
  },
};

export const noItemsMock = {
  ...layoutMock,
  items: [],
};

export const hasTaxonomiesMock = {
  ...layoutMock,
  currentLocation: 'yaml',
  currentOrganization: 'org1',
};

export const taxonomySwitcherProps = {
  organizations: organizations.available_organizations,
  locations: locations.available_locations,
  taxonomiesBool: { locations: true, organizations: true },
  isLoading: true,
};

export const userDropdownProps = {
  user: serverUser,
  notification_url: '/',
};

