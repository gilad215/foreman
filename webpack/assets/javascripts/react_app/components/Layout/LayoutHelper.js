import {
  changeOrganization,
  changeLocation,
  navigateTo,
} from '../../../foreman_navigation';

export const getCurrentPath = () => window.location.pathname;

export const combineMenuItems = (data) => {
  const items = [];

  data.menu.forEach((menu) => {
    menu.forEach((item) => {
      items.push(item);
    });
  });
  if (data.taxonomies.organizations) {
    items.push(createOrgItem(data.organizations.available_organizations));
  }

  if (data.taxonomies.locations) {
    items.push(createLocationItem(data.locations.available_locations));
  }
  return items;
};

const createOrgItem = (orgs) => {
  const anyOrg = {
    name: 'Any Organization',
    onClick: () => {
      changeOrganization('Any Organization');
      navigateTo('/organizations/clear');
    },
  };
  const childrenArray = [];
  childrenArray.push(anyOrg);

  orgs.forEach((org) => {
    const childObject = {
      type: org.type,
      name: org.title,
      onClick: () => {
        navigateTo(org.href);
        changeOrganization(org.title);
      },
      url: org.href,
    };
    childrenArray.push(childObject);
  });

  const orgItem = {
    type: 'sub_menu',
    name: 'Organizations',
    icon: 'fa fa-building',
    children: childrenArray,
    className: 'visible-xs-block',
    active: false,
  };
  return orgItem;
};

const createLocationItem = (locations) => {
  const anyLoc = {
    name: 'Any Location',
    onClick: () => {
      changeLocation('Any Location');
      navigateTo('/locations/clear');
    },
  };
  const childrenArray = [];
  childrenArray.push(anyLoc);

  locations.forEach((loc) => {
    const childObject = {
      type: loc.type,
      name: loc.title,
      onClick: () => {
        navigateTo(loc.href);
        changeLocation(loc.title);
      },
      url: loc.href,
    };
    childrenArray.push(childObject);
  });

  const locItem = {
    type: 'sub_menu',
    name: 'Locations',
    icon: 'fa fa-globe',
    children: childrenArray,
    className: 'visible-xs-block',
    active: false,
  };
  return locItem;
};
