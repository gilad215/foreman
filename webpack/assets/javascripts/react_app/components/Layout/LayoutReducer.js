import Immutable from 'seamless-immutable';
import { isEmpty } from 'lodash';

import {
  LAYOUT_SHOW_LOADING,
  LAYOUT_HIDE_LOADING,
  LAYOUT_CHANGE_ITEMS,
  LAYOUT_CHANGE_ORG,
  LAYOUT_CHANGE_LOCATION,
} from './LayoutConstants';

const initialState = Immutable({
  items: [],
  isLoading: false,
  activeMenu: '',
  currentOrganization: 'Any Organization',
  currentLocation: 'Any Location',
});

export default (state = initialState, action) => {
  const { payload } = action;

  switch (action.type) {
    case LAYOUT_SHOW_LOADING:
      return state.set('isLoading', true);

    case LAYOUT_HIDE_LOADING:
      return state.set('isLoading', false);

    case LAYOUT_CHANGE_ITEMS:
      if (!isEmpty(payload.active)) return state.set('items', payload.menuItems).set('activeMenu', payload.active);
      return state.set('items', payload.menuItems);

    case LAYOUT_CHANGE_ORG:
      return state.set('currentOrganization', payload.org);

    case LAYOUT_CHANGE_LOCATION:
      return state.set('currentLocation', payload.location);

    default:
      return state;
  }
};
