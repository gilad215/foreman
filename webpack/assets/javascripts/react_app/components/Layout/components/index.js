import React from 'react';
import { Loadable } from '../../../utils';

const loadingLayout = () => (
  <div id="layout">
    <div className="navbar navbar-pf-vertical-loading" />
    <div className="nav-pf-vertical nav-pf-vertical-with-sub-menus nav-pf-persistent-secondary" />
  </div>
);

export default Loadable(
  {
    loader: () => import('./Layout'),
  },
  loadingLayout
);
