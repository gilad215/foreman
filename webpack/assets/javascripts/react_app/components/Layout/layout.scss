@import "../../common/colors.scss";

.nav-pf-secondary-nav {
  background-color: $vertical-nav-highlight !important;

  .mobile-active {
    // Mark active Location/Org in Mobile view
    a {
      .list-group-item-value {
        background-color: $vertical-nav-background !important;
      }
    }
  }

  .list-group-item {
    a {
      background-color: $vertical-nav-highlight !important;

      &:hover {
        .list-group-item-value {
          background-color: $vertical-nav-background !important;
        }
      }
    }

    span {
      color: $pf-color-white !important;
    }
  }
}

.nav-pf-vertical {
  background-color: $vertical-nav-background !important;

  .list-group {
    .list-group-item {
      a {
        span {
          color: $vertical-nav-list-highlight;
        }

        &:hover {
          background-color: $vertical-nav-highlight;

          span {
            color: $pf-color-white;
          }
        }

        &::after {
          color: $vertical-nav-list-highlight;
        }
      }

      &:hover {
        a {
          background-color: $vertical-nav-highlight !important;

          span {
            color: $pf-color-white;
          }
        }
      }

      &.active {
        a {
          background-color: $vertical-nav-highlight;

          span {
            color: $pf-color-white;
          }
        }
      }
    }
  }
}

#vertical-spinner {
  margin-top: 15px;
  margin-left: 5px;
}

.navbar-header {
  width: 221.22px;
}
