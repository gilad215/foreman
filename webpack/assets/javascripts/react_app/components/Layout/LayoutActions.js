
import { isEmpty } from 'lodash';
import { combineMenuItems } from './LayoutHelper';
import { selectMenuItems } from './LayoutSelectors';
import {
  LAYOUT_SHOW_LOADING,
  LAYOUT_HIDE_LOADING,
  LAYOUT_UPDATE_ITEMS,
  LAYOUT_CHANGE_LOCATION,
  LAYOUT_CHANGE_ORG,
} from './LayoutConstants';

export const showLoading = () => ({
  type: LAYOUT_SHOW_LOADING,
});

export const hideLoading = () => ({
  type: LAYOUT_HIDE_LOADING,
});

export const changeActiveMenu = primary => (dispatch, getState) => {
  const menuItems = selectMenuItems(getState()).map(item =>
    (primary.title === item.name ? { ...item, active: true } : { ...item, active: false }));

  dispatch({
    type: LAYOUT_UPDATE_ITEMS,
    payload: {
      menuItems: isEmpty(menuItems) ? [] : menuItems,
      active: primary.title,
    },
  });
};

export const fetchMenuItems = data => (dispatch) => {
  const menuItems = combineMenuItems(data);
  dispatch({
    type: LAYOUT_UPDATE_ITEMS,
    payload: {
      menuItems: isEmpty(menuItems) ? [] : menuItems,
    },
  });
};

export const changeOrganization = org => (dispatch) => {
  dispatch({
    type: LAYOUT_CHANGE_ORG,
    payload: {
      org,
    },
  });
};

export const changeLocation = location => (dispatch) => {
  dispatch({
    type: LAYOUT_CHANGE_LOCATION,
    payload: {
      location,
    },
  });
};
