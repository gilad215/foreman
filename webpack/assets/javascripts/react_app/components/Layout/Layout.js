import React from 'react';
import PropTypes from 'prop-types';

import { VerticalNav } from 'patternfly-react';
import { isEmpty } from 'lodash';
import { noop } from '../../common/helpers';

import TaxonomySwitcher from './components/TaxonomySwitcher';
import UserDropdowns from './components/UserDropdowns';
import './layout.scss';

class Layout extends React.Component {
  componentDidMount() {
    const {
      items,
      data,
      fetchMenuItems,
      changeLocation,
      currentLocation,
      changeOrganization,
      currentOrganization,
    } = this.props;
    if (items.length === 0) fetchMenuItems(data);

    if (!isEmpty(data.locations.current_location)
     && currentLocation !== data.locations.current_location) {
      changeLocation(data.locations.current_location);
    }

    if (!isEmpty(data.organizations.current_org)
     && currentOrganization !== data.organizations.current_org) {
      changeOrganization(data.organizations.current_org);
    }
  }

  render() {
    const {
      items,
      data,
      isLoading,
      changeActiveMenu,
      changeOrganization,
      changeLocation,
      currentOrganization,
      currentLocation,
    } = this.props;
    return (
      <VerticalNav
        hoverDelay={0}
        items={items}
        onItemClick={changeActiveMenu}
        {...this.props}
      >
        <VerticalNav.Masthead>
          <VerticalNav.Brand
            title="foreman"
            iconImg={data.logo}
            href={data.root}
          />
          <TaxonomySwitcher
            taxonomiesBool={data.taxonomies}
            currentLocation={currentLocation}
            locations={data.locations.available_locations}
            onLocationClick={changeLocation}
            currentOrganization={currentOrganization}
            organizations={data.organizations.available_organizations}
            onOrgClick={changeOrganization}
            isLoading={isLoading}
          />
          <UserDropdowns
            notificationUrl={data.notification_url}
            user={data.user}
            changeActiveMenu={changeActiveMenu}
          />
        </VerticalNav.Masthead>
      </VerticalNav>
    );
  }
}

Layout.propTypes = {
  currentOrganization: PropTypes.string,
  currentLocation: PropTypes.string,
  isLoading: PropTypes.bool,
  fetchMenuItems: PropTypes.func,
  changeActiveMenu: PropTypes.func,
  changeOrganization: PropTypes.func,
  changeLocation: PropTypes.func,
  items: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    className: PropTypes.string,
    iconClass: PropTypes.string.isRequired,
    initialActive: PropTypes.bool,
    subItems: PropTypes.arrayOf(PropTypes.shape({
      title: PropTypes.string,
      isDivider: PropTypes.bool,
      className: PropTypes.string,
      onClick: PropTypes.func.isRequired,
    })),
  })),
  data: PropTypes.shape({
    menu: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.shape({
      type: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      icon: PropTypes.string.isRequired,
      children: PropTypes.any,
    }))),
    locations: PropTypes.shape({
      current_location: PropTypes.string,
      available_locations: PropTypes.arrayOf(PropTypes.shape({
        href: PropTypes.string.isRequired,
        id: PropTypes.number.isRequired,
        title: PropTypes.string,
      })),
    }),
    organizations: PropTypes.shape({
      current_organization: PropTypes.string,
      available_organizations: PropTypes.arrayOf(PropTypes.shape({
        href: PropTypes.string.isRequired,
        id: PropTypes.number.isRequired,
        title: PropTypes.string,
      })),
    }),
    root: PropTypes.string.isRequired,
    logo: PropTypes.string.isRequired,
    notification_url: PropTypes.string.isRequired,
    taxonomies: PropTypes.shape({
      locations: PropTypes.bool.isRequired,
      organizations: PropTypes.bool.isRequired,
    }),
    user: PropTypes.shape({
      current_user: PropTypes.object.isRequired,
      user_dropdown: PropTypes.arrayOf(PropTypes.shape({
        children: PropTypes.any,
        icon: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        type: PropTypes.string.isRequired,
      })),
    }),
  }),
};

Layout.defaultProps = {
  items: [],
  data: {},
  currentOrganization: 'Any Organization',
  currentLocation: 'Any Location',
  isLoading: false,
  fetchMenuItems: noop,
  changeActiveMenu: noop,
  changeOrganization: noop,
  changeLocation: noop,
};

export default Layout;
