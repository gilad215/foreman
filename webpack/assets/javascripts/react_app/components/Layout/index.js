import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from './LayoutActions';
import reducer from './LayoutReducer';
import { selectLayout, patternflyMenuItemsSelector } from './LayoutSelectors';

import Layout from './Layout';

// map state to props
const mapStateToProps = state => ({
  items: patternflyMenuItemsSelector(state),
  isLoading: selectLayout(state).isLoading,
  activeMenu: selectLayout(state).activeMenu,
  currentOrganization: selectLayout(state).currentOrganization,
  currentLocation: selectLayout(state).currentLocation,
});

// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export reducers
export const reducers = { layout: reducer };

// export connected component
export default connect(mapStateToProps, mapDispatchToProps)(Layout);
