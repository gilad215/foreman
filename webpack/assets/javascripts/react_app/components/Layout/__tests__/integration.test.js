import React from 'react';

import IntegrationTestHelper from '../../../common/IntegrationTestHelper';

import { hasTaxonomiesMock } from '../Layout.fixtures';
import Layout, { reducers } from '../index';

jest.mock('../../notifications/index', () => 'notification');

describe('Layout integration test', () => {
  it('should flow', async () => {
    const integrationTestHelper = new IntegrationTestHelper(reducers);

    const component = integrationTestHelper.mount(<Layout {...hasTaxonomiesMock} />);
    component.update();
    await IntegrationTestHelper.flushAllPromises();

    const yamlLocation = component.find('.location_menuitem');

    integrationTestHelper.takeStoreSnapshot('initial state');

    yamlLocation.at(0).simulate('click');

    integrationTestHelper.takeStoreAndLastActionSnapshot('Location "yaml" clicked');
    expect(component.find('#location-dropdown > .dropdown-toggle').text()).toBe('yaml');
  });
});
