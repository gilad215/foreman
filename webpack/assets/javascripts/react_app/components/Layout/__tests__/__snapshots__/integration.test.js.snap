// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location "yaml" clicked 1`] = `
Object {
  "action": Array [
    Array [
      Object {
        "payload": Object {
          "location": "yaml",
        },
        "type": "LAYOUT_CHANGE_LOCATION",
      },
    ],
  ],
  "state": Object {
    "layout": Object {
      "activeMenu": "",
      "currentLocation": "yaml",
      "currentOrganization": "org1",
      "isLoading": false,
      "items": Array [
        Object {
          "active": false,
          "children": Array [
            Object {
              "name": "Dashboard",
              "type": "item",
              "url": "/",
            },
            Object {
              "name": "Facts",
              "type": "item",
              "url": "/fact_values",
            },
          ],
          "icon": "fa fa-tachometer",
          "name": "Monitor",
          "type": "sub_menu",
        },
        Object {
          "active": false,
          "children": Array [
            Object {
              "name": "All Hosts",
              "type": "item",
              "url": "/hosts/new",
            },
            Object {
              "name": "Architectures",
              "type": "item",
              "url": "/architectures",
            },
          ],
          "icon": "fa fa-server",
          "name": "Hosts",
          "type": "sub_menu",
        },
        Object {
          "active": false,
          "children": Array [
            Object {
              "name": "Environments",
              "type": "item",
              "url": "/environments",
            },
            Object {
              "name": "Architectures",
              "type": "item",
              "url": "/architectures",
            },
          ],
          "icon": "fa fa-wrench",
          "name": "User",
          "type": "sub_menu",
        },
        Object {
          "active": false,
          "children": Array [
            Object {
              "name": "Domains",
              "type": "item",
              "url": "/domains",
            },
            Object {
              "name": "Realms",
              "type": "item",
              "url": "/realms",
            },
          ],
          "icon": "pficon pficon-network",
          "name": "Infrastructure",
          "type": "sub_menu",
        },
        Object {
          "active": false,
          "children": Array [
            Object {
              "name": "Any Organization",
              "onClick": [Function],
            },
            Object {
              "name": "org1",
              "onClick": [Function],
              "type": undefined,
              "url": "/organizations/1-org1/select",
            },
          ],
          "className": "visible-xs-block",
          "icon": "fa fa-building",
          "name": "Organizations",
          "type": "sub_menu",
        },
        Object {
          "active": false,
          "children": Array [
            Object {
              "name": "Any Location",
              "onClick": [Function],
            },
            Object {
              "name": "yaml",
              "onClick": [Function],
              "type": undefined,
              "url": "/locations/1-yaml/select",
            },
            Object {
              "name": "london",
              "onClick": [Function],
              "type": undefined,
              "url": "/locations/2-london/select",
            },
            Object {
              "name": "norway",
              "onClick": [Function],
              "type": undefined,
              "url": "/locations/3-norway/select",
            },
          ],
          "className": "visible-xs-block",
          "icon": "fa fa-globe",
          "name": "Locations",
          "type": "sub_menu",
        },
      ],
    },
  },
}
`;

exports[`initial state 1`] = `
Object {
  "layout": Object {
    "activeMenu": "",
    "currentLocation": "london",
    "currentOrganization": "org1",
    "isLoading": false,
    "items": Array [
      Object {
        "active": false,
        "children": Array [
          Object {
            "name": "Dashboard",
            "type": "item",
            "url": "/",
          },
          Object {
            "name": "Facts",
            "type": "item",
            "url": "/fact_values",
          },
        ],
        "icon": "fa fa-tachometer",
        "name": "Monitor",
        "type": "sub_menu",
      },
      Object {
        "active": false,
        "children": Array [
          Object {
            "name": "All Hosts",
            "type": "item",
            "url": "/hosts/new",
          },
          Object {
            "name": "Architectures",
            "type": "item",
            "url": "/architectures",
          },
        ],
        "icon": "fa fa-server",
        "name": "Hosts",
        "type": "sub_menu",
      },
      Object {
        "active": false,
        "children": Array [
          Object {
            "name": "Environments",
            "type": "item",
            "url": "/environments",
          },
          Object {
            "name": "Architectures",
            "type": "item",
            "url": "/architectures",
          },
        ],
        "icon": "fa fa-wrench",
        "name": "User",
        "type": "sub_menu",
      },
      Object {
        "active": false,
        "children": Array [
          Object {
            "name": "Domains",
            "type": "item",
            "url": "/domains",
          },
          Object {
            "name": "Realms",
            "type": "item",
            "url": "/realms",
          },
        ],
        "icon": "pficon pficon-network",
        "name": "Infrastructure",
        "type": "sub_menu",
      },
      Object {
        "active": false,
        "children": Array [
          Object {
            "name": "Any Organization",
            "onClick": [Function],
          },
          Object {
            "name": "org1",
            "onClick": [Function],
            "type": undefined,
            "url": "/organizations/1-org1/select",
          },
        ],
        "className": "visible-xs-block",
        "icon": "fa fa-building",
        "name": "Organizations",
        "type": "sub_menu",
      },
      Object {
        "active": false,
        "children": Array [
          Object {
            "name": "Any Location",
            "onClick": [Function],
          },
          Object {
            "name": "yaml",
            "onClick": [Function],
            "type": undefined,
            "url": "/locations/1-yaml/select",
          },
          Object {
            "name": "london",
            "onClick": [Function],
            "type": undefined,
            "url": "/locations/2-london/select",
          },
          Object {
            "name": "norway",
            "onClick": [Function],
            "type": undefined,
            "url": "/locations/3-norway/select",
          },
        ],
        "className": "visible-xs-block",
        "icon": "fa fa-globe",
        "name": "Locations",
        "type": "sub_menu",
      },
    ],
  },
}
`;
